controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +75,21 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new SQL query to fetch the total number of users is a good addition for pagination purposes. However, it's important to ensure that the database can handle this additional load without affecting performance. Consider adding an index on the `user_id`, `is_anonymous`, and `is_banned` fields if not already present.



<!-- This is an auto-generated comment by OSS CodeRabbit -->